api/models/logger.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 The Merlin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package models
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+
+	kfsv1alpha2 "github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2"
+)
+
+type LoggerMode string
+
+const (
+	LogAll      LoggerMode = "all"
+	LogRequest  LoggerMode = "request"
+	LogResponse LoggerMode = "response"
+)
+
+type Logger struct {
+	DestinationURL string        `json:"-"`
+	Model          *LoggerConfig `json:"model"`
+	Transformer    *LoggerConfig `json:"transformer"`
+}
+
+type LoggerConfig struct {
+	Enabled bool       `json:"enabled"`
+	Mode    LoggerMode `json:"mode"`
+}
+
+func GetLoggerMode(mode LoggerMode) kfsv1alpha2.LoggerMode {
+	modeMapping := map[LoggerMode]kfsv1alpha2.LoggerMode{
```

Comment:
nitpick: we can move this map object outside of the function so it will only initialize once